static/src/javascripts/projects/common/modules/gallery/lightbox.js

```diff
@@ -295,16 +297,97 @@ class GalleryLightbox {
         this.fsm.trigger(event, data);
     }
 
-    loadGalleryfromJson(galleryJson: GalleryJson, startIndex: number): void {
-        this.index = startIndex;
+    static loadNextOrPrevious(
+        currentImageId: string,
+        direction: string
+    ): Promise<Object> {
+        const pathPrefix = direction === 'forwards' ? 'getnext' : 'getprev';
+        const seriesTag = config
+            .get('page.nonKeywordTagIds')
+            .split(',')
+            .filter(tag => tag.includes('series'))[0];
+        const fetchUrl = `/${pathPrefix}/${seriesTag}/${currentImageId}`;
+        return fetch(fetchUrl)
+            .then(response => response.json())
+            .then(json =>
+                // filter out non-lightbox items in the series
+                json.filter(image => image.images.length > 0)
+            );
+    }
 
-        if (this.galleryJson && galleryJson.id === this.galleryJson.id) {
+    loadOrOpen(newGalleryJson: Object): void {
+        if (
+            this.galleryJson &&
+            newGalleryJson.id === this.galleryJson.id &&
+            newGalleryJson.images.length === this.galleryJson.images.length
+        ) {
             this.trigger('open');
         } else {
-            this.trigger('loadJson', galleryJson);
+            this.trigger('loadJson', newGalleryJson);
+        }
+    }
+
+    loadGalleryfromJson(
+        galleryJson: GalleryJson,
+        defaultIndex: number,
+        jumpToStartImage: boolean
+    ): void {
+        // if this is an image page, load series of images into the lightbox
+        if (
+            config.get('page.contentType') === 'ImageContent' &&
+            galleryJson.images.length < 2
+        ) {
+            // store current path with leading slash removed
+            const currentId = window.location.pathname.substring(1);
+            // fetch next and previous images and load them
+            Promise.all([
+                GalleryLightbox.loadNextOrPrevious(currentId, 'forwards'),
+                GalleryLightbox.loadNextOrPrevious(currentId, 'backwards'),
+            ])
+                .then(result => ({
```

Comment:
We could destructure this `result` array

.then(([nextResult, previousResult]) => ({
  next: nextResult.map(e => e.images[0])
  // ...
})